case-lib/lib.sh

```diff
@@ -810,9 +810,31 @@ is_firmware_file_zephyr()
     test "$znum" -gt 10
 }
 
+is_ipc4_debugfs()
+{
+    local ipc_type
+
+    ipc_file=/sys/kernel/debug/sof/ipc_type
+    ipc_type=$(cat $ipc_file) || {
+        return 1
+    }
+
+    if [ "$ipc_type" -eq 1 ]; then
+        return 0
+    fi
+    return 1
+}
+
 is_ipc4()
 {
     local ipc_type
+
+    # First try with debugfs
+    if is_ipc4_debugfs; then
+	return 0
+    fi
```

Comment:
```shell
if ipc_type=$(cat /sys/kernel/debug/sof/ipc_type); then
   # "cat" was successful
   case $ipc_type in
   0) return ... # in shell, success = error code 0 = "true"
   1) return ...
   *) die "unexpected /sys/kernel/debug/sof/ipc_type=$ipc_type"
fi

# "cat" failed, fallback on ...
```

Reply:
Good suggestion that anyone would understand, thanks.